### PR TITLE
Avoid potential dereference of a null pointer. Remove redundant call to get() on smart pointer.

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Comment by practicalswift** [comment]
@swift-ci please test

File: lib/Parse/ParseExpr.cpp [hunk 1]
```diff
@@ -836,9 +836,11 @@ static StringRef copyAndStripUnderscores(ASTContext &C, StringRef orig) {
   char *start = static_cast<char*>(C.Allocate(orig.size(), 1));
   char *p = start;
```
Comment (jtbandes) [inline lib/Parse/ParseExpr.cpp]: It doesn't look like `p` could ever be null here, unless allocation failed. `*p++` does not present a risk of `bool(p)` becoming false.

Comment (practicalswift) [inline lib/Parse/ParseExpr.cpp]: @jtbandes Yes, this issue is probably a bit theoretical but I'm trying to keep `clang-tidy` happy :-)

Comment (jtbandes) [inline lib/Parse/ParseExpr.cpp]: How about checking `p` after the allocation and not each time inside the loop?

Comment (practicalswift) [inline lib/Parse/ParseExpr.cpp]: @jtbandes Good idea! Fixed and pushed. Looks good?

**Review by jtbandes — APPROVED** [review]

**Comment by practicalswift** [comment]
@swift-ci please smoke test and merge

**Comment by practicalswift** [comment]
@swift-ci please test

**Comment by practicalswift** [comment]
@swift-ci please smoke test

**Pull request merged**